lib/orderbook/OrderBook.ts

```diff
@@ -345,6 +352,53 @@ class OrderBook extends EventEmitter {
     return outgoingOrder;
   }
 
+  /**
+   * Handles a request from a peer to create a swap deal. Checks if the order for the requested swap
+   * is available and if a route exists to determine if the request should be accepted or rejected.
+   * Responds to the peer with a swap response packet containing either an accepted quantity or rejection reason.
+   */
+  private handleSwapRequest = async (requestPacket: SwapRequestPacket, peer: Peer)  => {
```

Comment:
I don't think this function belongs into the orderbook.

Reply:
Previously it was in the Swaps module, but most of the logic to check whether a swap can be accepted is actually related to the order book. Namely checking if an order is valid, that it has sufficient quantity, and putting the accepted amount of the order on hold. I tried to leave the logic that had nothing to do with the order book in Swaps.

If you have a better idea let me know. I could maybe rename this method to something more specific.

Reply:
The thing is that the order book contains swaps.ts, not the other way around, so swaps is not aware of the orderbook and can't call one of its public methods without restructuring. I did spend some time thinking about this, and couldn't think of a better way. I do think the current approach makes sense, just like how the order book calls `beginSwap` when there's a match, it calls `acceptSwap` when there's a request to fill one of its orders. Let me know what you think.

Reply:
by contains, you mean a reference? what is wrong with two direction reference?  can't swaps be aware of orderbook?

Reply:
Yeah, it would be a circular reference/import. We create `swaps`, then create `orderbook` and pass `swaps` to its constructor, `orderbook` does not exist at the time `swaps` is created. Here's a good answer on reasons we'd want to avoid making them reference each other: https://softwareengineering.stackexchange.com/a/12030

An alternative would be to handle this in some higher level which makes calls to both `swaps` and `orderbook`, but I don't feel that would be better.